public/app/models/query/Condition.js

```diff
@@ -0,0 +1,84 @@
+// Based on http://kindohm.github.io/knockout-query-builder/js/condition.js
```

Comment:
Originally based on...  Actually there isn't much left of that original version, I've modified it a lot, but still recognising the source here is worthwhile.